serde-bson/src/main/java/io/micronaut/serde/bson/AbstractBsonMapper.java

```diff
@@ -176,10 +176,7 @@ private ByteBuffer toByteBuffer(InputStream inputStream) throws IOException {
         ReadableByteChannel channel = Channels.newChannel(inputStream);
         while (true) {
             int read = channel.read(byteBuffer);
-            if (read == 0) {
-                throw new IllegalStateException("Read only 0 bytes!");
-            }
-            if (read == -1) {
+            if (read <= 0) {
```

Comment:
is this the right thing to do? returning 0 from read is technically invalid, but it doesnt have to mean EOF necessarily

Reply:
I'm not sure, but without this check it loops forever since calling read again and again just always returns 0 if there is no more data

Reply:
ok, maybe it's a quirk of the bson streams then